projects/cli-files/go-cat/cmd/root_test.go

```diff
@@ -0,0 +1,57 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// cat [non-existent-file-path] displays error message : [non-existent-file-path]: no such file or directory
+func TestNoFileSpecified(t *testing.T) {
+	err := checkArgs("")
+	require.Equal(t, err.Error(), "go-cat: no file specified")
+}
+
+// cat [directory-path] displays error message : [directory-path] is a directory
+func TestDirectoryPath(t *testing.T) {
+	testCases := []struct {
+		name string
+		arg  string
+	}{{name: "current directory", arg: "."}, {name: "directory path", arg: "../../assets"}}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkArgs(tc.arg)
+			require.Equal(t, "go-cat: '"+tc.arg+"': is a directory", err.Error())
+		})
+	}
+}
+
+// cat [non-existent-file-path] displays error message : [non-existent-file-path]: no such file or directory
+func TestNonExistentFilePath(t *testing.T) {
+	err := checkArgs("/non-existent-file-path")
```

Comment:
While this is likely to work, it's _possible_ a computer you run this on may have a file at `/non-existent-file-path` - can you think of a way in your test you can make it _very very likely_ the path being used in the test doesn't exist, rather than just _quite unlikely_?

Reply:
Creating a temp file and delete it straight away :)

Reply:
That makes it a bit more unlikely :) But it's also possible some other test may create the same temp file in parallel. _Even more_ unlikely would be if you created a temp _directory_ (because when you create a directory, no one else can also create the same directory until you delete it), and pointed at a child of that that you didn't create (then delete the directory when you're done).